python_scripts/trees.py

```diff
@@ -320,20 +321,20 @@ def plot_decision_function(X, y, clf, ax=None):
 
 
 # %% [markdown]
-# As we visually assess, the partition defined by < 42 mm has mainly Adelie
-# penguin and only 2 samples which we could considered misclassified. However,
-# on the partition >= 42 mm, we cannot differentiate Gentoo and Chinstrap
-# (while they are almost twice more Gentoo).
+# As we visually assessed, the partition defined by < 42 mm has mainly Adelie
+# penguin and only 2 samples that are misclassified. However,
+# in the partition >= 42 mm, we cannot differentiate well between Gentoo and
+# Chinstrap (while they are almost twice more Gentoo).
 #
-# We should come with a statistical measure which combine the class
-# probabilities together that can be used as a criterion to qualify the purity
+# We should use a statistical measure that uses the all class
+# probabilities, as the criterion to qualify the purity
 # of a partition. We will choose as an example the entropy criterion (also used
 # in scikit-learn) which is one of the possible classification criterion.
 #
 # The entropy is defined as: $H(X) = - \sum_{k=1}^{K} p(X_k) \log p(X_k)$
 #
-# For a binary problem, the entropy function for one of the class can be
-# depicted as follows:
+# For a binary problem (e.g., only 2 classes of penguins), the entropy function
+# for one of the class can be depicted as follows:
 #
 # ![title](https://upload.wikimedia.org/wikipedia/commons/2/22/Binary_entropy_plot.svg)
 #
```

Comment:
(Below) Nitpick - I would explain it in terms of one class (as this is what is shown in the plot above) e.g., entropy max when proportion from the class is 50% (as other class is also 50%), min when only class x is present, or only the other class is present (prob=0)

Reply:
Also maybe a quick explanation of >2 classes, e.g., (I assume) for 3 classes entropy is max when all 3 classes are 33% ?

Reply:
In the code below I can read : 
> and minimum when only samples for a single class is present.

Do you think we should add more details here ?
With 3 classes, it is gonna be very verbose I think.

Reply:
Made a suggestion in new commit.